frontend/src/components/Login.jsx

```diff
@@ -1,89 +1,255 @@
+
+  const loginRequest = async (identifier, password) => {
+    const api_url = "http://0.0.0.0:8000/user/token";
+    const requestBody = queryString.stringify({
+      grant_type: "",
+      username: identifier,
+      password,
+      scope: "",
+      client_id: "",
+      client_secret: "",
+    });
+
+    try {
+      const response = await fetch(api_url, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/x-www-form-urlencoded",
+          accept: "application/json",
+        },
+        body: requestBody,
+      });
+
+      if (!response.ok) {
+        setLoginError(true);
+        setIdentifierValidated(true);
+        setPasswordValidated(true);
+        const data = await response.json();
+        if (data && data.detail) {
+          setLoginErrorMessage(data.detail);
+        } else {
+          setLoginErrorMessage("Something went wrong. Please try again later.");
         }
-    };
-
-    return (
-        <div className="login-container">
-            <form onSubmit={handleSubmit}>
-                <img src={logo} alt="Logo" className="logo_login" />
-                
-                <div className="password-container">
-                      <input 
-                          type="text" 
-                          placeholder="Usuari" 
-                          value={username} 
-                          onChange={e => setUsername(e.target.value)}
-                      />
-                </div>
-
-                <div className="password-container">
-                    <input 
-                        type={showPassword ? "text" : "password"} 
-                        placeholder="Contrasenya" 
-                        value={password} 
-                        onChange={e => setPassword(e.target.value)}
+      } else {
+        const data = await response.json();
+        setToken(data.access_token);
+        navigate("/userfeed");
+      }
+    } catch (error) {
+      setLoginError(true);
+      setLoginErrorMessage("Something went wrong. Please try again later.");
+    }
+  };
+
+  /* Events */
+
+  const onIdentifierChange = (e) => {
+    setIdentifier(e.target.value);
+    setIdentifierFilled(e.target.value.length > 0);
+    setIdentifierValidated(false);
+  };
+
+  const onPasswordChange = (e) => {
+    setPassword(e.target.value);
+    setPasswordFilled(e.target.value.length > 0);
+    setPasswordValidated(false);
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+    if (identifier.length == 0) {
+      setIdentifierFilled(false);
+      setIdentifierValidated(true);
+    }
+    if (password.length == 0) {
+      setPasswordFilled(false);
+      setPasswordValidated(true);
+    }
+
+    if (identifierFilled && passwordFilled) {
+      try {
+        loginRequest(identifier, password);
+      } catch (error) {
+        console.log(error);
+      }
+    }
+  };
+
+  return (
+    <div>
+      <Container className="py-5 min-vh-100">
```

Comment:
The utility classes used here are descriptive, providing padding and ensuring a minimum height. This enhances the user experience.